tests/30rooms/13anonymousaccess.pl

```diff
@@ -385,6 +385,109 @@ sub check_events
       })
    };
 
+test "GET /publicRooms lists rooms",
+   requires => [qw( first_api_client ), local_user_fixture() ],
+
+   check => sub {
+      my ( $http, $user ) = @_;
+
+      Future->needs_all(
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest0",
+         ),
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest1",
+         )->then( sub {
+            my ( $room_id ) = @_;
+
+            matrix_set_room_history_visibility( $user, $room_id, "world_readable" );
+         }),
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest2",
+         )->then( sub {
+            my ( $room_id ) = @_;
+
+            matrix_set_room_history_visibility( $user, $room_id, "invited" );
+         }),
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest3",
+         )->then( sub {
+            my ( $room_id ) = @_;
+
+            matrix_set_room_guest_access( $user, $room_id, "can_join" );
+         }),
+
+         matrix_create_room( $user,
+            visibility => "public",
+            room_alias_name => "listingtest4",
+         )->then( sub {
+            my ( $room_id ) = @_;
+
+            Future->needs_all(
+               matrix_set_room_guest_access( $user, $room_id, "can_join" ),
+               matrix_set_room_history_visibility( $user, $room_id, "world_readable" ),
+            );
+         }),
+      )->then( sub {
+         $http->do_request_json(
+            method => "GET",
+            uri    => "/api/v1/publicRooms",
+         )->then( sub {
```

Comment:
The point of `->then` chaining is that you don't have to indefinitely nest further and further in here. This can be

       $http->do_request(
          ...
       )
    })->then( sub {
       ...

and outdent the remainder of the test.